lib/stm32/common/spi_common_v12.c

```diff
@@ -0,0 +1,647 @@
+/** @addtogroup spi_file SPI peripheral API
+ * @ingroup peripheral_apis
+
+@author @htmlonly &copy; @endhtmlonly 2023
+Sebastien Lorquet <sebastien@lorquet.fr>
+
+Functions common to spiv1 and spiv2, but not usable in spiv3.
+*/
+
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2023 Sebastien Lorquet <sebastien@lorquet.fr>
```

Comment:
This is not fully correct. We need to include other contributors who almost got this correct.